newIDE/app/src/Profile/AuthenticatedUserProvider.js

```diff
@@ -875,17 +950,26 @@ export default class AuthenticatedUserProvider extends React.Component<
                   updateInProgress={this.state.editInProgress}
                 />
               )}
-            {this.state.emailVerificationPendingDialogOpen && (
-              <EmailVerificationPendingDialog
+            {this.state.emailVerificationDialogOpen && (
+              <EmailVerificationDialog
                 authenticatedUser={this.state.authenticatedUser}
                 onClose={() => {
-                  this.openEmailVerificationPendingDialog(false);
+                  this.openEmailVerificationDialog({
+                    open: false,
+                  });
                   this.state.authenticatedUser
                     .onRefreshFirebaseProfile()
                     .catch(() => {
                       // Ignore any error, we can't do much.
                     });
                 }}
+                sendEmailAutomatically={
+                  this.state.emailVerificationDialogProps.sendEmailAutomatically
+                }
+                showSendEmailButton={
+                  this.state.emailVerificationDialogProps.showSendEmailButton
```

Comment:
Should you just spread the props with `{...this.state.emailVerificationDialogProps}`?